pkg/scheduler/internal/queue/scheduling_queue.go

```diff
@@ -383,9 +385,10 @@ func NewPriorityQueue(
 		moveRequestCycle:                  -1,
 		isSchedulingQueueHintEnabled:      utilfeature.DefaultFeatureGate.Enabled(features.SchedulerQueueingHints),
 	}
-	pq.cond.L = &pq.lock
+	pq.cond.L = &pq.qLock
```

Comment:
`pq.qLock` is the same as `pq.activeQLock` in this pr #125692 . It would be better to standardize the lock name.
https://github.com/kubernetes/kubernetes/blob/31e89b1f4de112b3795611711e288ba100e981db/pkg/scheduler/internal/queue/scheduling_queue.go#L401

Reply:
It's not the same. After these two PRs, there will be 3 locks in scheduling queue: `nominatorLock` for protecting nominator, `activeQLock` for protecting activeQ related operations and `qLock` for the rest of the logic.

In #125692 the `cond` lock was changed to  `activeQLock` as activeQ relies on it.

Reply:
Gotcha. 
I'm not sure how many places will use `qLock` after these two pr megerd. Just curious about the performance difference if we just use two locks(`activeQlock` can replace `qLock`, ) instead of three.

Reply:
My point is the more locks there are, the more maintenance cost.

Reply:
The goal of these two PRs is to decouple scheduling cycles from queueing. `qLock` which is the general lock in the scheduling queue is used almost all the time there, especially when running pre enqueue and scheduling hints plugins that can be time consuming. Before the #125692, `lock` management resulted in blocking `Pop()` operation what led to scheduling throughput degradation, especially for the many gated pods case. Adding `activeQLock` allows us to limit locking time and decouples `Pop()` operation without losing consistency, by protecting only the activeQ related operations. Nevertheless, introducing the `activeQLock` is not sufficient, as the second bottleneck were two operations on `nominator` inside the scheduling cycle, which still required decoupling. It ended in introducing a yet another lock - `nominatorLock` in this PR.

After these PRs, the `qLock` will still protect most of the scheduling queue. The `activeQLock` and `nominatorLock` are only supportive in the places where decoupling from scheduling cycle is required. 

We understand the maintenance cost, but keeping the performant scheduler is critical, especially if there is no better solution found that won't result in redesigning the scheduling queue.

Reply:
> but keeping the performant scheduler is critical

Agree. Performance is most significant.